lib/server.js

```diff
@@ -8,27 +8,36 @@ module.exports = {
 
   createServer (config) {
     const server = new Hapi.Server()
-    server.connection({
+    const defaultOptions = {
       host: config.web.host,
       port: config.web.port,
       routes: {
         files: {
           relativeTo: config.main.paths.root
         }
       }
-    })
+    }
+
+    server.connection(_.defaultsDeep(config.web.options || {}, defaultOptions))
 
     return server
   },
 
   registerPlugins (app, server) {
-    return new Promise((resolve, reject) => {
-      server.register(app.config.web.plugins || [ ], err => {
-        if (err) return reject(err)
+    const promises = _.map(app.config.web.plugins || [], plugin => {
+      return new Promise((resolve, reject) => {
+        server.register(_.omit(plugin, ['onLoad']), err => {
+          if (_.isFunction(plugin.onLoad))
+            plugin.onLoad.call(server, err)
```

Comment:
Nope It's my custom property. And seems that I was wrong with it. 
It's much easier to add a one callback that will be invoked after all plugins loaded. 

is it ok if I'll add a `web.onPluginLoaded` property ?